### DISCUSSION TITLE
Client/server should exchange library version and/or protocol versions

### DISCUSSION THREAD

**Discussion by brycekahle** [sockjs/sockjs-client#557] [Ideas]
Currently the client and server don't communicate what library version and protocol version they are running/support. This information should be communicated, at a minimum from server to client, so the client can fail early if version compatibility is not possible.

**Use cases**
1. Supporting multiple client library/protocol versions by running multiple servers and changing `base_url` in the `info` response to match the client. Server work might be needed for this logic to be possible.
2. Ensuring client and server use compatible protocol versions.
3. Ensuring iframes serve up the matching client library version url. Extension of use case 1.

**Server to Client**

Easily accomplished by adding version(s) to `info` response.

**Client to Server**

Logical place seems in the `info` request. Options are:
1. Adding a query string version. E.g. `?v=1.0.0-beta.6`. This has caching implications.
2. Changing from GET to POST for the `info` request.

**Comment by jkarneges** [comment]
I wonder if there's a way the iframe code could be made generic such that it doesn't need to match with the client version? Maybe it can delegate most of the logic to the main window instead. Or maybe the main window could send its own code to the iframe after initialization.
